### DISCUSSION TITLE
Reading mime contents of .msg file type attachments

### DISCUSSION THREAD

**Discussion by kamalvk18** [ecederstrand/exchangelib#1256] [General]
Hi, Thanks for the support.

I'm trying to load the attachments from .msg files into the azure blob storage. Whenever there are attachments of excel or pdf types I'm able to get the data from the attachments as bytes and I'm able to upload the file to blob storage.

 But Whenever there is a .msg type attachment. I'm not able to get the mime_content as bytes, I'm getting a Message object instead. Is there any way I can get the mime_content as bytes?

Any help on this will be much appreciated!

Thank you.

**Comment by ecederstrand** [comment]
It's a bit hard to follow what you're trying to do. Can you please post the code you use to read the .msg files?

**Reply by kamalvk18** [reply]
`
    for filename in os.listdir(directory):
        msg = extract_msg.openMsg(os.path.join(directory, filename))

    attachments = msg.attachments
    for attachment in attachments:
        if attachment.name.endswith('.xlsb'):
            print(attachment.name, attachment.data[:100])
        if attachment.name.endswith('.msg'):
            print(attachment.name, attachment.data)
`

As you can see in the below screenshot, Output for excel type attachments is bytes. But for .msg type attachments it is Message object.
<img width="900" alt="image" src="https://github.com/ecederstrand/exchangelib/assets/69080406/3acc0841-1052-4a1c-b393-2e5d7785a733">

But Instead I want mime_content of the .msg attachment as bytes. Is there a way I can achieve this?
